pkg/resource/managed_reconciler.go

```diff
@@ -360,6 +372,21 @@ func (r *ManagedReconciler) Reconcile(req reconcile.Request) (reconcile.Result,
 		return reconcile.Result{RequeueAfter: r.shortWait}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
 	}
 
+	if !IsConditionTrue(managed.GetCondition(v1alpha1.TypeReferencesResolved)) {
```

Comment:
Per our discussion this morning one down side of gating reference resolution on a condition being set is that we'll only ever process the resolution once. In the current implementation if someone edits a managed resource and changes a reference that corresponds to a mutable field, we'll ignore it.

I was thinking we could avoid this by updating the `AttributeReferencer` implementations to be no-ops if the fields they would assign were already set. This would avoid the need to check the `TypeReferencesResolved` condition, but would have the same issue in that it wouldn't support updating references after creation time.

Perhaps we should just run reference resolution every time we reconcile the managed resource? Definitely happy to defer this to a future PR in the interest of getting this iteration merged.

Reply:
I agree with the direction, but I think this should be deferred to another pr. I opened https://github.com/crossplaneio/crossplane-runtime/issues/53 to track this.